gocourse10/observer/observer.go

```diff
@@ -0,0 +1,35 @@
+package observer
+
+type Observer interface {
+	TimeToEat()
+	GetID() int
+}
+
+type Item struct {
+	ObserverList []Observer
+}
+
+func (i *Item) Register(o Observer) {
+	i.ObserverList = append(i.ObserverList, o)
+}
+
+func (i *Item) Deregister(o Observer) {
+	i.ObserverList = removeFromslice(i.ObserverList, o)
```

Comment:
Оскільки слайси ми вже давно пройшли, пропоную не ускладнювати та використати функцію [`slices.DeleteFunc`](https://pkg.go.dev/slices#DeleteFunc):
```suggestion
	i.ObserverList = slices.DeleteFunc(i.ObserverList, func(e Observer) bool {
		return o.GetID() == e.GetID()
	})
```

Reply:
А `removeFromslice` видалити.